src/App.tsx

```diff
@@ -1,24 +1,148 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { Todolist } from './components/Todolist/Todolist';
+import { Footer } from './components/Footer/Footer';
+import { Filter } from './types/Filter';
+import { Errors } from './types/Error';
 
-const USER_ID = 0;
+const USER_ID = 11843;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(Filter.all);
+  const [error, setError] = useState('');
+  const [activeFilter, setActiveFilter] = useState(Filter.all);
+  const [temptodo, setTempTodo] = useState<Todo | null>(null);
+  const [cleared, setcleared] = useState<boolean>(false);
+  const [toggled, setToggled] = useState<string>('');
+  const titleField = useRef<HTMLInputElement | null>(null);
+
+  const clickHandler = (value: Filter) => {
+    switch (value) {
+      case Filter.active:
+        setFilterBy(Filter.active);
+        setActiveFilter(Filter.active);
```

Comment:
Why do you need two exactly the same filter states?

Reply:
They are different, and needed for different purposes